combat-trainer.lic

```diff
@@ -2648,41 +2648,44 @@ class TrainerProcess
       reset_ability(game_state, 'Meraud')
       return
     end
-    if /meraud/i =~ bput('commune sense', 'roundtime', 'Meraud', 'Meraud\'s')
+    if /meraud/i =~ DRC.bput('commune sense', 'roundtime', 'Meraud', 'Meraud\'s')
       reset_ability(game_state, 'Meraud', 60)
       game_state.blessed_room = true
       return
     end
     waitrt?
     @equipment_manager.stow_weapon(game_state.weapon_name)
-    fput "get #{@water_holder} from my #{@theurgy_supply_container}"
-    if /that/i =~ bput("sprinkle #{@water_holder} on #{checkname}", 'You sprinkle', 'Sprinkle what?', 'Sprinkle that')
-      fput "put #{@water_holder} in my #{@theurgy_supply_container}"
+    unless DRCI.exists?(@water_holder, @theurgy_supply_container,)
+      DRC.message("Cannot get #{@water_holder}, skipping Meraud commune")
       @equipment_manager.wield_weapon?(game_state.weapon_name, game_state.weapon_skill)
       @training_abilities.delete('Meraud')
       return
     end
-    bput "put #{@water_holder} in my #{@theurgy_supply_container}", 'You put', 'What were you referring to'
-    fput "get #{@flint_lighter}", 'You get a', 'You are already'
-    if bput("get incense from my #{@theurgy_supply_container}", 'You get', 'You are already', 'referring to') == 'referring to'
-      fput "stow #{@flint_lighter}", 'You put your', 'Stow what'
+    unless DRCTH.has_holy_water?(@theurgy_supply_container, @water_holder)
+      DRC.message("Cannot get holy water, skipping Meraud commune")
       @equipment_manager.wield_weapon?(game_state.weapon_name, game_state.weapon_skill)
       @training_abilities.delete('Meraud')
       return
     end
-    fput 'light my incense with flint' # get messaging
-    pause
-    waitrt?
-    fput "wave incense at #{checkname}"
-    fput 'snuff incense'
-    bput("put incense in my #{@theurgy_supply_container}", 'You put', 'What were you referring to')
-    fput "stow #{@flint_lighter}", 'You put your', 'Stow what'
+    unless DRCTH.sprinkle_holy_water(@theurgy_supply_container, @water_holder, checkname)
```

Comment:
So this does return a bool, but it's only the result of putting away the holy water (not getting or sprinkling). Maybe that's ok, or maybe `sprinkle_holy_water` needs to be changed. What do you think?

Reply:
To put that differently, there's some redundancy/duplication in these three checks.

[DRCI.exists?](https://github.com/rpherbig/dr-scripts/blob/master/common-items.lic#L390) taps the water holder to make sure it's in the container.

[DRCTH.has_holy_water?](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L40) gets the water holder from the container (which is already a predicate function), checks if there is water inside, then puts the water holder away.

[DRCTH.sprinkle_holy_water](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L145) get the water holder form the container (again), calls `sprinkle?`, then puts the water holder away (again).

-----

I feel like what we really want here is "were we able to sprinkle the holy water?", which is _almost_ what `sprinkle_holy_water` does on its own. My only concern is that `sprinkle_holy_water` returns whether or not the item was put away, _not_ whether or not we could sprinkle the holy water.

Now, all of that said, `sprinkle_holy_water` is only used in two places in `commune.lic`, so I think what I'd like is to make a `sprinkle_holy_water?` which returns a more meaningful value and use it.

What do you think about that?

Reply:
Actually, I found a bug in  [[DRCTH.has_holy_water?](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L40)](https://github.com/rpherbig/dr-scripts/blob/master/common-theurgy.lic#L40) for the case where you can't get the water holder, but you can tap the water inside it. The function returns true, even though you can't get the holder to sprinkle yourself with. I'm finding a few weird odds and ends with common-theurgy, and more as I look.

Let me try and clean this up, but also work on common-theurgy on the side.